amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/oas/OasDocumentParser.scala

```diff
@@ -306,31 +308,32 @@ abstract class OasDocumentParser(root: Root)(implicit val ctx: OasWebApiContext)
     api
   }
 
-  def parseSecurity(entry: YMapEntry, api: WebApi) = {
-    entry.value.tagType match {
+  def parseSecurity(entry: YMapEntry, api: WebApi): Unit = {
+    val requirements: Seq[SecurityRequirement] = entry.value.tagType match {
       case YType.Seq =>
         val idCounter = new IdCounter()
-        val securedBy =
-          entry.value
-            .as[Seq[YNode]]
-            .map(s => OasLikeSecurityRequirementParser(s, api.withSecurity, idCounter).parse()) // todo when generating id for security requirements webapi id is null
-            .collect { case Some(s) => s }
-      //api.set(WebApiModel.Security, AmfArray(securedBy, Annotations(entry.value)))
+        entry.value
+          .as[Seq[YNode]]
+          .flatMap(s =>
+            OasLikeSecurityRequirementParser(s, (se: SecurityRequirement) => se.adopted(api.id), idCounter)
+              .parse()) // todo when generating id for security requirements webapi id is null
       case _ =>
         ctx.eh.violation(InvalidSecurityRequirementsSeq,
                          entry.value,
                          "'security' must be an array of security requirement object")
+        Nil
     }
+    val extension: Seq[SecurityRequirement] = api.security
+    api.set(WebApiModel.Security, AmfArray(requirements ++ extension, Annotations(entry.value)), Annotations(entry))
   }
 
   private def parseEndpoints(api: WebApi, entry: YMapEntry) = {
     val paths = entry.value.as[YMap]
     val endpoints =
       paths
         .regex("^/.*")
-        .foldLeft(List[EndPoint]())((acc, curr) =>
-          acc ++ ctx.factory.endPointParser(curr, api.withEndPoint, acc).parse())
-    if (endpoints.nonEmpty) api.set(WebApiModel.EndPoints, AmfArray(endpoints), Annotations(entry.value))
+        .foldLeft(List[EndPoint]())((acc, curr) => acc ++ ctx.factory.endPointParser(curr, api.id, acc).parse())
+    api.set(WebApiModel.EndPoints, AmfArray(endpoints, Annotations(entry.value)), Annotations(entry))
```

Comment:
We shouldn't add the Endpoints field if the `endpoints` map isn't present.

Reply:
This code only can be reach when the key "paths" is present. If paths is present, then we use that entry as source maps of the endpoints field. If path key is present, endpoints field needs to be set. Is not the same have empty paths that not have the key at all.